impl/src/main/java/org/jboss/weld/executor/DaemonThreadFactory.java

```diff
@@ -44,4 +44,9 @@ public Thread newThread(Runnable r) {
         thread.setDaemon(true);
         return thread;
     }
+
+    // Holder class to postpone thread group creation until when it's needed.
```

Comment:
What's the gain here? I understand that you want a static reference to the group, but how does postponing the creation help?  That thread group is referenced from `AbstractExecutorServices` and as such will be created anyway during any bootstrap. What am I missing? :)

Reply:
The thought was that the thread group is not going to be created until someone instantiates FixedThreadPoolExecutorServices or TimingOutFixedThreadPoolExecutorServices. If it's a simple static field the thread is going to be created already when somebody just references the DaemonThreadFactory class, not even having to instantiate it.

There is probably not going to be a lot of difference at the end of the day, I'm OK to update this however you prefer.